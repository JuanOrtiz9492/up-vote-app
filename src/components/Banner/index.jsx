import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

const BannerContainer = styled.div`
    background: ${props=> props.background};
    display: grid;
    grid-template-columns: ${props => props.template};
    margin: ${props => props.margin};
    padding: ${props => props.padding};
    width: 100%;
    position: relative;
    box-sizing: border-box;
`

const Banner = ({
    background,
    template,
    margin,
    padding,
    children,
    }) => (
        <BannerContainer
            background = {background}
            template = {template}
            margin = {margin}
            padding = {padding}
        >
            {children}
        </BannerContainer>
    );

Banner.propTypes = {
    background: PropTypes.string,
    template: PropTypes.string,
    margin: PropTypes.string,
    children: PropTypes.element,
    padding: PropTypes.element,
};
    
Banner.defaultProps = {
    background: '#ffffff',
    template: '100%',
    margin: '0',
    children: null,
    margin: '0'
}

export default Banner;