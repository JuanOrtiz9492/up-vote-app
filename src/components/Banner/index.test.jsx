import React from 'react';
import { render } from '@testing-library/react';
import Banner from '.';

describe('Banner Component', () => {

    it('should Match the Snapshot', () => {
        const { container } = render(<Banner />);
        expect(container).toMatchSnapshot();
      })
})
