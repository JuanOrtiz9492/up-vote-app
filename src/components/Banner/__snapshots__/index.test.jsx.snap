// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner Component should Match the Snapshot 1`] = `
<div>
  <div
    class="sc-AykKC dJdDiY"
  />
</div>
`;
